File: Algorithms/p386_Lexicographical_Numbers_1.cpp
class Solution {
public:
    vector<int> lexicalOrder(int n) {
        vector<int> res(n);
        int i = 0;
        int m = 1;
        while(i<n){
            res[i++] = m;
            m *= 10;
            while(m>n){
                m /= 10;
                ++m;
                while(m%10==0) m /= 10;
            }
        }
        return res;
    }
};

File: Algorithms/p386_Lexicographical_Numbers/p386_Lexicographical_Numbers_1.py
class Solution(object):
    def lexicalOrder(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        res = [None]*n
        i = 0
        m = 1
        while i<n:
            res[i] = m
            i += 1
            m *= 10
            while m>n:
                m //= 10
                m += 1
                while m%10==0:
                    m //= 10
        return res
